File: 0225_easy_hot_implement-stack-using-queues/solution.java
import java.util.LinkedList;
import java.util.Queue;

class MyStack {
    Queue<Integer> mainQueue;
    Queue<Integer> tempQueue;

    public MyStack() {
        this.mainQueue = new LinkedList<>();
        this.tempQueue = new LinkedList<>();
    }
    
    public void push(int x) {
        this.mainQueue.offer(x);
    }
    
    public int pop() {
        int size = this.mainQueue.size();
        for( int i = 0; i < size - 1; i++) {
            this.tempQueue.offer(this.mainQueue.poll());
        }

        int result = this.mainQueue.poll();

        Queue<Integer> t = this.mainQueue;
        this.mainQueue = this.tempQueue;
        this.tempQueue = t;

        return result;
        
    }
    
    public int top() {
        int size = this.mainQueue.size();
        for( int i = 0; i < size - 1; i++) {
            this.tempQueue.offer(this.mainQueue.poll());
        }

        int result = this.mainQueue.peek();

        this.tempQueue.offer(this.mainQueue.poll());

        Queue<Integer> t = this.mainQueue;
        this.mainQueue = this.tempQueue;
        this.tempQueue = t;

        return result;
    }
    
    public boolean empty() {
        return this.mainQueue.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */

File: 0225_easy_hot_implement-stack-using-queues/solution.py
from collections import deque


class MyStack:

    def __init__(self):
        self.main_queue = deque()
        self.temp_queue = deque()
        
    def push(self, x: int) -> None:
        self.main_queue.append(x)

    def pop(self) -> int:
        if len(self.main_queue) == 0:
            return None
        
        for _ in range(len(self.main_queue) - 1):
            self.temp_queue.append(self.main_queue.popleft())
        
        result = self.main_queue.popleft()

        self.main_queue, self.temp_queue = self.temp_queue, self.main_queue
        
        return result

    def top(self) -> int:
        if len(self.main_queue) == 0:
            return None
        
        for _ in range(len(self.main_queue) - 1):
            self.temp_queue.append(self.main_queue.popleft())
        
        result = self.main_queue[0]
        self.temp_queue.append(self.main_queue.popleft())

        self.main_queue, self.temp_queue = self.temp_queue, self.main_queue

        return result
        
    def empty(self) -> bool:
        return len(self.main_queue) == 0 and len(self.temp_queue) == 0
        


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()